app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListFragment.kt

```diff
@@ -367,12 +381,23 @@ class ThreadListFragment : Fragment(), SwipeRefreshLayout.OnRefreshListener {
                 folderRole == FolderRole.ARCHIVE
             }
             SwipeAction.DELETE -> {
-                val shouldKeepItem = when (folderRole) {
-                    FolderRole.DRAFT, FolderRole.SPAM, FolderRole.TRASH -> true
-                    else -> false
-                }
-                deleteWithConfirmationPopup(folderRole, count = 1) { deleteThread(thread.uid) }
-                shouldKeepItem
+                deleteWithConfirmationPopup(
+                    folderRole = folderRole,
+                    count = 1,
+                    displayLoader = false,
+                    onDismiss = {
+                        // Notify only if the user cancelled the popup (e.g. the thread is not deleted),
+                        // otherwise it will notify the next item in the list and make it slightly blink
+                        if (threadListAdapter.dataSet.indexOfFirst { it is Thread && it.uid == thread.uid } == position) {
```

Comment:
Why do we need the condition? Can we not just `notifyItemChanged(position)` if `onDismiss` triggered?

Reply:
No we can't, because onDismiss is called if we delete a thread, thus notifying will redraw the new position (the thread after the one deleted), then the api call will return, and we will redraw it once more, so it will blink

Reply:
Ah yeah dismiss also triggers when positive button is clicked, it's not the first time it surprises me. Then, can we replace `indexOfFirst` and the complicated predicate with `indexOf`?